exporters/memory/include/opentelemetry/exporters/memory/in_memory_span_exporter.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "opentelemetry/sdk/trace/exporter.h"
+#include "opentelemetry/sdk/trace/span_data.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace memory
+{
+/**
+ * A mock exporter that switches a flag once a valid recordable was received.
+ */
+class InMemorySpanExporter final : public opentelemetry::sdk::trace::SpanExporter
+{
+public:
+  InMemorySpanExporter(
+      std::shared_ptr<std::vector<std::unique_ptr<sdk::trace::SpanData>>> spans_received,
+      std::shared_ptr<bool> shutdown_called = std::shared_ptr<bool>(new bool(false)));
+
+  std::unique_ptr<sdk::trace::Recordable> MakeRecordable() noexcept override;
+
+  sdk::trace::ExportResult Export(
+      const nostd::span<std::unique_ptr<sdk::trace::Recordable>> &recordables) noexcept override;
+
+  void Shutdown(std::chrono::microseconds timeout = std::chrono::microseconds(0)) noexcept override;
+
+private:
+  std::shared_ptr<std::vector<std::unique_ptr<sdk::trace::SpanData>>> spans_received_;
+  std::shared_ptr<bool> shutdown_called_;
```

Comment:
This shouldn't need to be a shared_ptr. Bools take up negligible memory and can be passed by value, so using a shared_ptr just adds complexity.

Reply:
Updated to use a normal bool and added a getter fucntion

Reply:
Actually, this update caused some tests to fail. Adding a normal bool w/ a getter function won't work because the tests rely on SpanExporter (a parent class of all exporters), and the added getter function is not a standard function between all exporters.

Would you recommend I keep the current approach, add a public bool, or try a different approach?

Reply:
Removed shutdown_called_ field, as it was only used for testing, as per Oulin's recommendation